tests/integrations/gcp/test_gcp.py

```diff
@@ -0,0 +1,320 @@
+                .functions()
+                .generateUploadUrl(parent=parent)
+            )
+            upload_url_response = api_request.execute()
+
+            upload_url = upload_url_response.get("uploadUrl")
+
+            # STEP : Upload zip file of cloud function to generated signed url
+            with open(os.path.join(tmpdir, "ball.zip"), "rb") as data:
+                requests.put(
+                    upload_url,
+                    data=data,
+                    headers={
+                        "x-goog-content-length-range": "0,104857600",
+                        "content-type": "application/zip",
+                    },
+                )
+
+        # STEP : Create a new cloud function
+        location = "projects/{}/locations/{}".format(project_id, location_id)
+
+        function_url = "https://{}-{}.cloudfunctions.net/{}".format(
+            location_id, project_id, function_name
+        )
+
+        body = {
+            "name": name,
+            "description": "Created as part of testsuite for getsentry/sentry-python",
+            "entryPoint": "cloud_handler",
+            "runtime": functions_runtime,
+            "timeout": timeout,
+            "availableMemoryMb": 128,
+            "sourceUploadUrl": upload_url,
+            "httpsTrigger": {"url": function_url},
+        }
+
+        api_request = (
+            functions_service.projects()
+            .locations()
+            .functions()
+            .create(location=location, body=body)
+        )
+        api_request.execute()
+
+        # STEP : Invoke the cloud function
+        # Adding delay of 60 seconds for new created function to get deployed.
+        time.sleep(60)
+        api_request = (
+            functions_service.projects().locations().functions().call(name=name)
+        )
+        function_call_response = api_request.execute()
+
+        # STEP : Fetch logs of invoked function
+        log_name = "projects/{}/logs/cloudfunctions.googleapis.com%2Fcloud-functions".format(
+            project_id
+        )
+        project_name = "projects/{}".format(project_id)
+        body = {"resourceNames": [project_name], "filter": log_name}
+
+        log_service = build("logging", "v2", credentials=creds)
+
+        api_request = log_service.entries().list(body=body)
+        log_response = api_request.execute()
+
+        for entry in log_response.get("entries", []):
+            entry_log_name = entry.get("logName")
+            entry_function_name = (
+                entry.get("resource", {}).get("labels", {}).get("function_name")
+            )
+            entry_text_payload = entry.get("textPayload", "")
+            if (
+                entry_log_name == log_name
+                and entry_function_name == function_name
+                and "EVENTS: " in entry_text_payload
+            ):
+                event = entry_text_payload[len("EVENTS: ") :]
+                events.append(json.loads(event))
+
+        log_flag = True
```

Comment:
This flag never gets set to `False` so I think it cannot serve a purpose.

Reply:
Sorry, this was a minor mistake, I'll correct it, On line no. 255 it actually needs to be set to false.